File: corpus/java/Negative.java
/**
 * Negative control. Mentions cryptography, uses none.
 * Historical note: we removed MD5 and DES from this service in 2020.
 */
public class Negative {
    private static final String AES_KEY_ENV = "APP_AES_KEY";
    private static final int RSA_MIN_BITS = 2048;

    public String policy() {
        return "Reject RSA below 2048 and any DES or RC4 cipher suite.";
    }

    public void describe() {
        System.out.println("SHA-256 is our standard digest");
    }

    public boolean isDeserialized(Object o) {
        return o != null;
    }
}

File: corpus/go/negative.go
package main

// Negative control. Mentions algorithms in prose and identifiers only.
// We removed md5 and des usage from this package in 2021.

const (
	rsaMinBits   = 2048
	aesKeyEnvVar = "APP_AES_KEY"
)

func policy() string {
	return "Reject RSA below 2048; DES and RC4 are not permitted."
}

func describe(name string) string {
	return "configured digest: " + name
}

func addresses(in []string) []string {
	return in
}
